libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,194 @@
+    await this.fetchAll(newContext);
+  }
+
+  // if the cache is active, or we're connected, or streaming is disabled, we're ready.
+  get ready() {
+    return this._connected;
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+  ): ResolutionDetails<boolean> {
+    return this.evaluate(flagKey, 'boolValue');
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+  ): ResolutionDetails<string> {
+    return this.evaluate(flagKey, 'stringValue');
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+  ): ResolutionDetails<number> {
+    return this.evaluate(flagKey, 'doubleValue');
+  }
+
+  resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+  ): ResolutionDetails<U> {
+    return this.evaluate(flagKey, 'objectValue');
+  }
+
+  private evaluate<T extends FlagValue>(
+    flagKey: string,
+    type: AnyFlagResolutionType,
+  ): ResolutionDetails<T> {
+    const resolved = this._flags[flagKey];
+    if (!resolved) {
+      throw new FlagNotFoundError(`flag key ${flagKey} not found in cache`);
+    }
+    if (resolved.type !== type) {
+      throw new TypeMismatchError(`flag key ${flagKey} is not of type ${type}`);
+    }
+    return {
+      reason: this._connected ? resolved.reason : StandardResolutionReasons.CACHED,
+      variant: resolved.variant,
+      value: resolved.value as T
+    }
+  }
+
+  private async retryConnect(context: EvaluationContext) {
+    this._delayMs = Math.min(this._delayMs * BACK_OFF_MULTIPLIER, this._maxDelay);
+    if (this._retry >= this._maxRetries) {
+      this._logger?.warn(`${FlagdWebProvider.name}: max retries reached`);
+      return;
+    }
+    this._callbackClient.eventStream(
+      {},
+      (message) => {
+        this._logger?.debug(`${FlagdWebProvider.name}: event received: ${message.type}`);
+        switch (message.type) {
+          case EVENT_PROVIDER_READY:
+            this.fetchAll(context).then(() => {
+              this.resetConnectionState();
+              window.dispatchEvent(new Event(ProviderEvents.Ready));
+            });
+            return;
+          case EVENT_CONFIGURATION_CHANGE: {
+            this.fetchAll(context).then(() => {
+              window.dispatchEvent(new Event(ProviderEvents.ConfigurationChanged));
+            })
+            return;
+          }
+        }
+      },
+      () => {
+        window.dispatchEvent(new Event(ProviderEvents.Error));
+        this._logger?.error(`${FlagdWebProvider.name}: could not establish connection to flagd`);
+        setTimeout(() => this.retryConnect(context), this._delayMs);
+      }
+    );
+    this._retry++;
+  }
+
+  private async fetchAll(context: EvaluationContext) {
+    const transformedContext = this.transformContext(context);
+    const allResolved = await this._promiseClient.resolveAll({ context: transformedContext });
```

Comment:
Should anything happen if the call to `resolveAll` fails? Perhaps we should retry a few times? Is there a timeout?

Reply:
Should the existing flags be marked as stale until flagd responds with the updated values?